app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,39 @@
+class CommentsController < ApplicationController
+  def index
```

Comment:
only methods for create, edit ,update and delete should be here.
comments on certain post should be retrieved with post.comments.all